lib/src/str_util.rs

```diff
@@ -78,6 +89,15 @@ impl StringPattern {
         }
     }
 
+    /// Returns the original string of this pattern.
+    pub fn as_str(&self) -> &str {
+        match self {
+            StringPattern::Exact(literal) => literal,
+            StringPattern::Glob(pattern) => pattern.as_str(),
+            StringPattern::Substring(needle) => needle,
+        }
+    }
```

Comment:
Do we not want to include the kind prefix (e.g. "exact:")?

Reply:
I think it would have to be handled by caller. For revset usage, the pattern string gets already unquoted/unescaped, so we can't reconstruct the original string. For CLI usage, it wouldn't be ideal to add `exact:` to bare pattern.

Perhaps, we can add `kind_str()` method if needed.

Reply:
Makes sense.